src/kvstore/NebulaStore.h

```diff
@@ -28,28 +36,37 @@ struct SpacePartInfo {
 class NebulaStore : public KVStore, public Handler {
     FRIEND_TEST(NebulaStoreTest, SimpleTest);
     FRIEND_TEST(NebulaStoreTest, PartsTest);
+    FRIEND_TEST(NebulaStoreTest, ThreeCopiesTest);
 
 public:
     NebulaStore(KVOptions options,
                 std::shared_ptr<folly::IOThreadPoolExecutor> ioPool,
-                std::shared_ptr<thread::GenericThreadPool> workers,
                 HostAddr serviceAddr)
             : ioPool_(ioPool)
-            , workers_(workers)
             , storeSvcAddr_(serviceAddr)
             , raftAddr_(getRaftAddr(serviceAddr))
             , options_(std::move(options)) {
         partMan_ = std::move(options_.partMan_);
         init();
     }
 
-    ~NebulaStore() = default;
+    ~NebulaStore();
 
     // Calculate the raft service address based on the storage service address
     static HostAddr getRaftAddr(HostAddr srvcAddr) {
+        if (srvcAddr == HostAddr(0, 0)) {
+            return srvcAddr;
+        }
         return HostAddr(srvcAddr.first, srvcAddr.second + 1);
```

Comment:
+1 or -1 is an implicit dependency. mark a comment.

Reply:
Good point.